waspc/cli/src/Wasp/Cli/GithubRepo.hs

```diff
@@ -0,0 +1,129 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Wasp.Cli.GithubRepo where
+
+import qualified Codec.Archive.Tar as Tar
+import qualified Codec.Compression.GZip as GZip
+import Control.Exception (SomeException, try)
+import Data.Aeson
+  ( FromJSON,
+    parseJSON,
+    withObject,
+    (.:),
+  )
+import qualified Data.ByteString.Lazy as BL
+import Data.Functor ((<&>))
+import Data.List (intercalate)
+import Data.Maybe (fromJust, maybeToList)
+import Network.HTTP.Conduit (simpleHttp)
+import qualified Network.HTTP.Simple as HTTP
+import Path.IO (copyDirRecur, removeDirRecur)
+import StrongPath (Abs, Dir, File, Path', (</>))
+import qualified StrongPath as SP
+import StrongPath.Path (toPathAbsDir)
+import System.Directory (createDirectoryIfMissing)
+import Wasp.Cli.FileSystem (getUserCacheDir)
+
+type GithubRepo = (GithubRepoOwner, GithubRepoName)
+
+type GithubRepoOwner = String
+
+type GithubRepoName = String
+
+fetchFolderFromGithubRepoToDisk ::
+  forall d.
+  GithubRepo ->
+  String ->
```

Comment:
So `pathToFolderInRepo` is path but you used `String`.
If I got it right, we know it is relative path to directory -> I believe I saw you using `SP.parseRelDir` in one place, so obviously you expect it to be that.

You actually use it in two places total:
1. In first place you expect it to be rel dir path on the disk. So we can do `Path' Rel' Dir'`.
2. In second place you need it to be Posix, because you are using it URL. Also relative, and also dir. So `Path Posix Rel' Dir'`.

So I would suggest that in fetchFolderFromGithubRepoToDisk you say it is `Path Posix Rel' Dir'`, or maybe you can use `Path Posix (Rel a) (Dir b)` to give whoever is passing this URL a bit more freedom. But it should be Posix, because it is `pathToFolderInRepo`, online repo, so URLs are the thing online, and Posix usually makes more sense then.

Then, in your first place where you use it, you can do `relDirToSystem`, and later in the second place you can do `fromRelDirP`, since you need posix there.

Ah crap I just realized I don't have `relDirToSystem` in StrongPath hm. I have `relDirToPosix`, but I don't have `relDirToWindows` or `relDirToSystem` (ther is issue for those on strong path gh repo). The trick is you can easily switch rel path from one standard to another because you just need to swap all separators with correct separator.

Well, there are a couple of options then:
1. Update StrongPath to have `relDirToSystem` :D (but you also need to publish it which you can't yeah, blah).
2. Implement it here in Utils or somewhere, by importing StrongPath.Internal, I think that should work. Later we can transfer it to actual StrongPath library.
3. Type it as `Path System (Rel a) (Dir b)` for now, and turn it into Posix in that place where you need Posix. Leave a TODO saying that this should semantically be better to be typed as Posix, but well at the end it doesn't make a big difference.

If any of this stuff is a bit confusing, ask @sodic for help, he has seen enough StrongPath to become a StrongPath expert.

Reply:
I've named it `targetRepoFolderName` now and it is representing a folder name.

Reply:
I've updated the function to accept a StrongPath